### DISCUSSION TITLE
Restrict new user logins

### DISCUSSION THREAD

**Discussion by NiklasRosenstein** [outline/outline#2592] [Self Hosting]
Hi! I've set up Outline with Google OIDC, and it works well. I want to use it as a personal note taking app. The problem I have is that everyone can now log into my Outline instance. As a workaround to making my collections quasi-publicly available, I've set the default permissions to "No Access", and that seems to work fine.

Yet, I would prefer if I had some more control over who can can access my Outline instance from the Outline configuration. The Google OIDC is super convenient to get family members and friends onboarded, so I'd also like to keep Google as the IdP.

Does outline have a feature to block sign-in for users that have never signed in before, or allow only sign in for users with email addresses from an allowlist, or only if they received an Outline invite?

Thanks for this great piece of software :)

Cheers,

**Comment by tommoor** [comment]
Hi Niklas, the only functionality that currently exists in this vein is the allowed domains:

https://github.com/outline/outline/blob/main/.env.sample#L131-L133

**Reply by jfmontanaro** [reply]
Is the allowed-domains setting evaluated on every login, or just when creating a user? If the latter, then is it possible to simply set allowed-domains to null (or a nonsense value in case null means "everything") after all desired users have been created?

**Reply by tommoor** [reply]
It's only used during team creation, basically it exists to restrict to a single Google team originally. I think the scope of this variable could be expanded to check against all user signups but that's not the case today.

**Reply by NiklasRosenstein** [reply]
Thanks for the response @tommoor. Unfortunately that won't help here at all because everyone logging in through Google OICP will either have a `gmail.com` or `googlemail.com` domain. 😅 Would you be open to a contribution to configure an allowlist for full email addresses or an option to only accept pre-registered/invited email addresses?
